### PR TITLE
[enhancement](compaction) Control the parallelism for urgent compaction tasks

### PR DESCRIPTION
**Pull request by Hastyshell**
## Proposed changes

For some urgent compaction tasks, their submittion should take parallelism into account.

Currently, we apply the control policy for data loading in specific. Other source of urgent tasks are considered as eager.

### CODE REVIEW

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run p0

File: be/src/olap/olap_server.cpp [hunk 1]
```diff
@@ -1097,7 +1097,35 @@ Status StorageEngine::_submit_compaction_task(TabletSharedPtr tablet,
...
+               compaction_type == CompactionType::CUMULATIVE_COMPACTION);
+        std::map<DataDir*, std::unordered_set<TabletSharedPtr>> copied_cumu_map;
+        std::map<DataDir*, std::unordered_set<TabletSharedPtr>> copied_base_map;
+        {
+            std::unique_lock<std::mutex> lock(_tablet_submitted_compaction_mutex);
+            copied_cumu_map = _tablet_submitted_cumu_compaction;
+            copied_base_map = _tablet_submitted_base_compaction;
+        }
+        auto stores = get_stores();
+        bool is_busy = std::none_of(
```
Comment (gavinchou) [inline be/src/olap/olap_server.cpp:1112]: too many indentions

**Comment by TangSiyang2001** [comment]
run buildall

**Review by luwei16 — APPROVED** [review]

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**